service/history/handler.go

```diff
@@ -206,6 +209,22 @@ func (h *Handler) isStopped() bool {
 	return atomic.LoadInt32(&h.status) == common.DaemonStatusStopped
 }
 
+func (h *Handler) DeepHealthCheck(
+	_ context.Context,
+	_ *historyservice.DeepHealthCheckRequest,
+) (_ *historyservice.DeepHealthCheckResponse, retError error) {
+	defer log.CapturePanic(h.logger, &retError)
+	h.startWG.Wait()
+
+	latency := h.persistenceHealthSignal.AverageLatency()
+	errRatio := h.persistenceHealthSignal.ErrorRatio()
+
+	if latency > h.config.HealthPersistenceLatencyFailure() || errRatio > h.config.HealthPersistenceErrorRatio() {
```

Comment:
stats maybe unreliable on service restart. not sure if we want to take host age into account.

Reply:
I think this StartupMembershipJoinDelay already done the job. I will see if I still need this.